src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php

```diff
@@ -0,0 +1,392 @@
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Security\Annotation\DemoRestricted;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * ZoneController is responsible for handling "Improve > International > Locations > Zones"
+ */
+class ZoneController extends FrameworkBundleAdminController
+{
+    /**
+     * Show all zones.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param ZoneFilters $zoneFilters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, ZoneFilters $zoneFilters): Response
+    {
+        $zoneGridFactory = $this->get('prestashop.core.grid.factory.zone');
+        $zoneGrid = $zoneGridFactory->getGrid($zoneFilters);
+
+        return $this->render('@PrestaShop/Admin/Improve/International/Zone/index.html.twig', [
+            'zoneGrid' => $this->presentGrid($zoneGrid),
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'layoutHeaderToolbarBtn' => $this->getZoneToolbarButtons(),
+        ]);
+    }
+
+    /**
+     * Provides filters functionality.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request): RedirectResponse
+    {
+        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
+
+        return $responseBuilder->buildSearchResponse(
+            $this->get('prestashop.core.grid.definition.factory.zone'),
+            $request,
+            ZoneGridDefinitionFactory::GRID_ID,
+            'admin_zones_index'
+        );
+    }
+
+    /**
+     * Show "Add new" zone form and handles its submit.
+     *
+     * @AdminSecurity(
+     *     "is_granted('create', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_zones_index",
+     *     message="You need permission to create new zone."
+     * )
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request): Response
+    {
+        $zoneFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.zone_form_builder');
+        $zoneFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.zone_form_handler');
+
+        $zoneForm = $zoneFormBuilder->getForm();
+        $zoneForm->handleRequest($request);
+
+        try {
+            $handleResult = $zoneFormHandler->handle($zoneForm);
+
+            if (null !== $handleResult->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful creation', 'Admin.Notifications.Success'));
```

Comment:
```suggestion
                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
```

Reply:
Hi @boubkerbribri,
We try to use punctuation only on full sentences, e.g. 'The item was successfully deleted." but not on short sentences such as "Successful creation", "Successful deletion" that are not really full sentences. 
So I'd rather keep 'Successful creation" here. :-)